src/goto-instrument/nondet_static.cpp

```diff
@@ -10,14 +12,53 @@ Date: November 2011
 \*******************************************************************/
 
 /// \file
-/// Nondeterministic initialization of certain global scope variables
+/// Nondeterministic initialization of global scope variables
 
 #include "nondet_static.h"
 
 #include <goto-programs/goto_model.h>
 
 #include <linking/static_lifetime_init.h>
 
+/// See the return.
+/// \param sym The symbol expression to analyze.
+/// \param ns Namespace for resolving type information
+/// \return True if the symbol expression holds a static symbol which can be
+/// nondeterministically initialized, false otherwise.
+bool is_nondet_initializable_static(
+  const symbol_exprt &sym,
+  const namespacet &ns)
+{
+  const irep_idt &id = sym.get_identifier();
+
+  // is it a __CPROVER_* variable?
+  if(has_prefix(id2string(id), CPROVER_PREFIX))
+    return false;
+
+  // variable not in symbol table such as symex variable?
+  if(!ns.get_symbol_table().has_symbol(id))
+    return false;
+
+  // any other internal variable such as Java specific?
```

Comment:
Would the comment be better saying something like "is the type explicitly marked as not to be nondet initialized?" I assume other language front-ends could conceivably also want to mark some types in the same way, so it might be nice not to make this sound too Java centric.

Reply:
I agree - this is not necessarily Java specific.